packages/bot-skeleton/src/scratch/dbot.js

```diff
@@ -39,27 +41,32 @@ class DBot {
             this.workspace.cached_xml = { main: main_xml, toolbox: toolbox_xml };
             Blockly.derivWorkspace = this.workspace;
 
+            this.workspace.addChangeListener(event => saveWorkspaceToRecent(save_types.UNSAVED, event));
             this.workspace.addChangeListener(this.valueInputLimitationsListener.bind(this));
-            this.workspace.addChangeListener(event => updateDisabledBlocks(this.workspace, event));
+            this.workspace.addChangeListener(updateDisabledBlocks);
             this.addBeforeRunFunction(this.unselectBlocks.bind(this));
             this.addBeforeRunFunction(this.disableStrayBlocks.bind(this));
             this.addBeforeRunFunction(this.checkForErroredBlocks.bind(this));
             this.addBeforeRunFunction(this.checkForRequiredBlocks.bind(this));
 
             // Push main.xml to workspace and reset the undo stack.
-            Blockly.Xml.domToWorkspace(Blockly.Xml.textToDom(main_xml), this.workspace);
+            const recent_files = getSavedWorkspaces();
+            this.workspace.currentStrategy = Blockly.utils.genUid();
```

Comment:
- Could we rename this to `current_strategy_id`? 
- Please use snake_case for properties.

Reply:
We always use camelCase for Blockly property. Do we wan to make it inconsistent?